### PR TITLE
[SPARK-8095] Resolve dependencies of --packages in local ivy cache

### PR DESCRIPTION
**Pull request by brkyvz**
Dependencies of artifacts in the local ivy cache were not being resolved properly. The dependencies were not being picked up. Now they should be.

cc @andrewor14

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala [hunk 1]
```diff
@@ -112,26 +112,32 @@ class SparkSubmitUtilsSuite extends SparkFunSuite with BeforeAndAfterAll {
   }
 
   test("search for artifact at local repositories") {
-    val main = new MavenCoordinate("my.awesome.lib", "mylib", "0.1")
+    val main = new MavenCoordinate("my.great.lib", "mylib", "0.1")
+    val dep = "my.great.dep:mydep:0.5"
```
Comment (brkyvz) [inline core/src/test/scala/org/apache/spark/deploy/SparkSubmitUtilsSuite.scala]: Added this dependency as a regression test

File: core/src/test/scala/org/apache/spark/deploy/IvyTestUtils.scala [hunk 2]
```diff
@@ -256,6 +320,7 @@ private[deploy] object IvyTestUtils {
       } else {
         FileUtils.deleteDirectory(repo)
       }
+      FileUtils.deleteDirectory(new File(ivySettings.getDefaultCache, artifact.groupId))
```
Comment (brkyvz) [inline core/src/test/scala/org/apache/spark/deploy/IvyTestUtils.scala]: We need to delete the cache for these fake libraries, or they affect each other between tests.

**Comment by andrewor14** [comment]
retest this please

**Comment by andrewor14** [comment]
retest this please

**Comment by andrewor14** [comment]
@brkyvz there seems to a legitimate test failure

**Comment by andrewor14** [comment]
retest this please

**Comment by andrewor14** [comment]
retest this please

**Comment by andrewor14** [comment]
Merging into master 1.4 thanks Burak

**Comment by andrewor14** [comment]
@brkyvz could you open a patch against branch-1.3? It doesn't merge cleanly there.

**Pull request closed without merging**